examples/shouldi/shouldi/dependency_check.py

```diff
@@ -0,0 +1,72 @@
+import json
+import os
+import tempfile
+from pathlib import Path
+import asyncio
+from typing import Dict, Any
+
+from dffml.df.base import op
+from dffml.df.types import Definition
+
+package_src_dir = Definition(name="package_src_dir", primitive="str")
+dependency_check_output = Definition(
+    name="dependency_check_output", primitive="Dict[str, Any]"
+)
+
+
+@op(
+    inputs={"pkg": package_src_dir},
+    outputs={"report": dependency_check_output},
+)
+async def run_dependency_check(pkg: str) -> Dict[str, Any]:
+    """
+    CLI usage: dffml service dev run -log debug shouldi.dependency_check:run_dependency_check -pkg .
+    """
+    print(os.getcwd())
+    with tempfile.TemporaryDirectory() as tempdir:
+        if Path(pkg).is_file():
+            proc = await asyncio.create_subprocess_exec(
+                "dependency-check.sh",
+                "-f",
+                "JSON",
+                "--out",
+                os.path.abspath(tempdir),
+                "-s",
+                pkg,
+                cwd=os.path.dirname(pkg),
+                stdout=asyncio.subprocess.PIPE,
+                stderr=asyncio.subprocess.PIPE,
+            )
+        else:
+            proc = await asyncio.create_subprocess_exec(
+                "dependency-check.sh",
+                "-f",
+                "JSON",
+                "--out",
+                os.path.abspath(tempdir),
+                "-s",
+                ".",
+                cwd=pkg,
+                stdout=asyncio.subprocess.PIPE,
+                stderr=asyncio.subprocess.PIPE,
+            )
+
```

Comment:
It looks like in a139539e92d7ec67264d043eb776eaf9e6161bc9 we lost the call to `communicate`, without that call we don't wait for the process to complete. Since the process hasn't finished yet, there's no `.json` file.

```suggestion

       _, stderr = await proc.communicate()
       if proc.returncode != 0:
           raise DependencyCheckError(stderr.decode())

```